### PR TITLE
Change plot for juicy queenbeet (#1)

### PR DESCRIPTION
**Pull request by mrpandat**
Change plot for juicy queenbeet, to be accessible at lvl 7

Did some refacto about how you visualise your plot in code, hope you like it, maybe would be nice to do it for every recipe.

Result:
image

### CODE REVIEW

**Comment by Chakaa** [comment]
Hi,
That's indeed a nice way to put a design in the code, however, the existing plot for juicy queenbeet should be kept for level 9+.
Indeed, your level 7 setup is nice when you do not have any issue, but having the double walls allows for a single seed to die, and only affect 1 mutation spot. Here, if your center seed die, then all 4 spots become useless.

With that level 9 setup, I'll gladly merge your branch.

**Comment by mrpandat** [comment]
> 
> 
> Hi, That's indeed a nice way to put a design in the code, however, the existing plot for juicy queenbeet should be kept for level 9+. Indeed, your level 7 setup is nice when you do not have any issue, but having the double walls allows for a single seed to die, and only affect 1 mutation spot. Here, if your center seed die, then all 4 spots become useless.
> 
> With that level 9 setup, I'll gladly merge your branch.

Thanks for your answer, glad you like the design in the code i made !
When I'll have some time, i'll do another PR with some heavy refactoring on all seeds recipes with this design.

I'll finish the PR this week :)

File: CookieGardenHelper-reloaded/main.js [hunk 1]
```diff
@@ -624,56 +624,67 @@ Game.registerMod("cookiegardenhelperreloaded",{
...
-				[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],
-				[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],
-				[[0,0],[p2,0],[p2,0],[p2,0],[p2,0],[0,0]],
-				[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],
-				[[0,0],[p1,0],[p1,0],[p1,0],[p1,0],[0,0]],
-				[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]]
+				[X,X,X,X,X,X],
+				[X,X,X,X,X,X],
+				[X,X,X,X,X,X],
+				[X,X,Z,Z,Z,Z],
```
Comment (Chakaa) [inline CookieGardenHelper-reloaded/main.js:676]:
```suggestion
				[X,Z,Z,Z,Z,X],

```

**Review by Chakaa — COMMENTED** [review]
Thank you but some setup does not follow CC setup for the farm, I've suggested the correct version, please have a look.

File: CookieGardenHelper-reloaded/main.js [hunk 2]
```diff
@@ -624,56 +624,67 @@ Game.registerMod("cookiegardenhelperreloaded",{
...
 			];
 		}else if(l>=4){
 			return [
-				[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],
-				[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],
-				[[0,0],[p2,0],[p2,0],[p2,0],[p2,0],[0,0]],
-				[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],
-				[[0,0],[p1,0],[p1,0],[p1,0],[p1,0],[0,0]],
-				[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]]
+				[X,X,X,X,X,X],
```
[the remark this answers is not in the document]
Comment (mrpandat) [reply CookieGardenHelper-reloaded/main.js:673]: done

File: CookieGardenHelper-reloaded/main.js [hunk 3]
```diff
@@ -624,56 +624,67 @@ Game.registerMod("cookiegardenhelperreloaded",{
...
 			];
 		}else if(l>=3){
 			return [
-				[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],
-				[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],
-				[[0,0],[0,0],[p2,0],[p2,0],[p2,0],[0,0]],
-				[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],
-				[[0,0],[0,0],[p1,0],[p1,0],[p1,0],[0,0]],
-				[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]]
+				[X,X,X,X,X,X],
```
[the remark this answers is not in the document]
Comment (mrpandat) [reply CookieGardenHelper-reloaded/main.js:682]: done

File: CookieGardenHelper-reloaded/main.js [hunk 4]
```diff
@@ -692,42 +703,133 @@ Game.registerMod("cookiegardenhelperreloaded",{
...
+					[X,Q,Q,Q,Q,Q],
+					[X,Q,X,Q,X,Q],
+					[X,Q,Q,Q,Q,Q],
+					[X,Q,X,Q,X,Q],
+					[X,Q,Q,Q,Q,Q]
+				];
 			}
 			if(l>=3){
-				return [[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],[[0,0],[0,0],[21,0],[21,0],[21,0],[0,0]],[[0,0],[0,0],[21,0],[0,0],[21,0],[0,0]],[[0,0],[0,0],[21,0],[21,0],[21,0],[0,0]],[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]]];
+				return [
```
[the remark this answers is not in the document]
Comment (mrpandat) [reply CookieGardenHelper-reloaded/main.js:729]: done

File: CookieGardenHelper-reloaded/main.js [hunk 5]
```diff
@@ -692,42 +703,133 @@ Game.registerMod("cookiegardenhelperreloaded",{
...
+					[X,C,X,C,X,C],
+					[X,X,X,X,X,X],
+					[X,C,X,C,X,C],
+					[X,C,X,C,X,C]
+				];
 			}
 			else if(l>=6){
-				return [[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],[[0,0],[5,0],[5,0],[0,0],[5,0],[5,0]],[[0,0],[0,0],[5,0],[0,0],[5,0],[0,0]],[[0,0],[0,0],[5,0],[0,0],[5,0],[0,0]],[[0,0],[5,0],[5,0],[0,0],[5,0],[5,0]],[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]]];
+				return [	
+					[X,X,X,X,X,X],
```
[the remark this answers is not in the document]
Comment (mrpandat) [reply CookieGardenHelper-reloaded/main.js:776]: done

File: CookieGardenHelper-reloaded/main.js [hunk 6]
```diff
@@ -692,42 +703,133 @@ Game.registerMod("cookiegardenhelperreloaded",{
...
+					[X,C,C,X,C,C],
+					[X,X,C,X,C,X],
+					[X,X,C,X,C,X],
+					[X,C,C,X,C,C]
+				];
 			}
 			else if(l>=5){
-				return [[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],[[0,0],[5,0],[0,0],[5,0],[5,0],[0,0]],[[0,0],[5,0],[0,0],[5,0],[0,0],[0,0]],[[0,0],[5,0],[0,0],[5,0],[0,0],[0,0]],[[0,0],[5,0],[0,0],[5,0],[5,0],[0,0]],[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]]];
+				return [
```
[the remark this answers is not in the document]
Comment (mrpandat) [reply CookieGardenHelper-reloaded/main.js:785]: done

File: CookieGardenHelper-reloaded/main.js [hunk 7]
```diff
@@ -692,42 +703,133 @@ Game.registerMod("cookiegardenhelperreloaded",{
...
+					[X,X,C,X,C,C],
+					[X,X,C,X,C,X],
+					[X,X,C,X,C,X],
+					[X,X,C,X,C,C]
+				];
 			}
 			else if(l>=4){
-				return [[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],[[0,0],[5,0],[0,0],[5,0],[5,0],[0,0]],[[0,0],[5,0],[0,0],[5,0],[0,0],[0,0]],[[0,0],[5,0],[0,0],[5,0],[0,0],[0,0]],[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]]];
+				return [	
+					[X,X,X,X,X,X],
```
[the remark this answers is not in the document]
Comment (mrpandat) [reply CookieGardenHelper-reloaded/main.js:796]: done

File: CookieGardenHelper-reloaded/main.js [hunk 8]
```diff
@@ -692,42 +703,133 @@ Game.registerMod("cookiegardenhelperreloaded",{
...
+					[X,X,X,C,X,C],
+					[X,X,X,C,X,C],
+					[X,X,X,C,X,C]
+				];
 			}
 			else if(l>=2){
-				return [[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],[[0,0],[0,0],[5,0],[0,0],[5,0],[0,0]],[[0,0],[0,0],[5,0],[0,0],[5,0],[0,0]],[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]]];
+				return [	
+					[X,X,X,X,X,X],
```
[the remark this answers is not in the document]
Comment (mrpandat) [reply CookieGardenHelper-reloaded/main.js:816]: done

File: CookieGardenHelper-reloaded/main.js [hunk 9]
```diff
@@ -692,42 +703,133 @@ Game.registerMod("cookiegardenhelperreloaded",{
...
+					[X,X,X,X,X,X],
+					[X,X,X,X,X,X],
+					[X,X,X,C,X,C],
+					[X,X,X,C,X,C]
+				];
 			}
 			else if(l>=1){
-				return [[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],[[0,0],[0,0],[5,0],[0,0],[0,0],[0,0]],[[0,0],[0,0],[5,0],[0,0],[0,0],[0,0]],[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]]];
+				return [
```
[the remark this answers is not in the document]
Comment (mrpandat) [reply CookieGardenHelper-reloaded/main.js:825]: done

File: CookieGardenHelper-reloaded/main.js [hunk 10]
```diff
@@ -692,42 +703,133 @@ Game.registerMod("cookiegardenhelperreloaded",{
...
+					[X,X,X,X,X,X],
+					[X,X,C,X,C,C],
+					[X,X,C,X,C,X],
+					[X,X,C,X,C,X]
+				];
 			}
 			else if(l>=3){
-				return [[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],[[0,0],[0,0],[5,0],[0,0],[5,0],[0,0]],[[0,0],[0,0],[5,0],[0,0],[5,0],[0,0]],[[0,0],[0,0],[5,0],[0,0],[5,0],[0,0]],[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]]];
+				return [	
+					[X,X,X,X,X,X],
```
[the remark this answers is not in the document]
Comment (mrpandat) [reply CookieGardenHelper-reloaded/main.js:806]: done

**Comment by mrpandat** [comment]
> Thank you but some setup does not follow CC setup for the farm, I've suggested the correct version, please have a look.

Indeed, thanks for your vigilance, I have done the changes you asked for. 👍

**Pull request merged**